addons/web/static/src/js/services/notification_service.js

```diff
@@ -82,8 +85,25 @@ var NotificationService = AbstractService.extend({
         }
         var NotificationWidget = params.Notification || Notification;
         var notification = this.notifications[++id] = new NotificationWidget(this, params);
-        notification.appendTo(this.$el);
-        return id;
+        if (params.await) {
+            return notification.appendTo(this.$el).then(function () {
+                return id;
+            });
+        } else {
+            notification.appendTo(this.$el);
+            return id;
+        }
+    },
+    /**
+     * It may sometimes be useful to get the notification element. For
+     * example, when part of the notification content needs to be updated such
+     * as a progress bar.
+     *
+     * @param {number} notificationId
+     * @returns {jQuery}
+     */
+    getElement: function (notificationId) {
+        return this.notifications[notificationId].$el;
```

Comment:
Probably not needed, I guess the progress bar is rendered somewhere then given to the notification? Why can't you keep a reference to that progressbar instead of having to find it again inside the notification later?

Reply:
Done as discussed, see `contentEl` param